docs/sources/developers/plugins/add-support-for-variables.md

```diff
@@ -58,3 +58,97 @@ For data sources, you need to use the [getTemplateSrv]({{< relref "../../package
      return { data };
    }
    ```
+
+## Add support for variable queries to your data source
+
+[Query variables]({{< relref "../../variables/variable-types/add-query-variable/" >}}) is a type of variable that allows you to query a data source for the values. By adding support for variable queries to your data source plugin, users can create dynamic dashboards based on data from your data source.
+
+Let's start by defining a query model for the variable query.
+
+```ts
+export interface MyVariableQuery {
+  namespace: string;
+  rawQuery: string;
+}
+```
+
+For a data source to support query variables you need to override the [`metricFindQuery`]({{< relref "../../packages_api/data/datasourceapi.md#metricfindquery-method" >}}) in your `DataSourceApi` class. `metricFindQuery` returns an array of [`MetricFindValue`]({{< relref "../../packages_api/data/metricfindvalue.md" >}}) which has a single property, `text`.
+
+```ts
+async metricFindQuery(query: MyVariableQuery, options?: any) {
+  // Retrieve DataQueryResponse based on query.
+  const response = await this.doQuery(query);
```

Comment:
Maybe we can highlight using `query` as an object, because in case of string query we don't need custom editor.
```suggestion
  const response = await this.fetchMetricNames(query.namespace, query.rawQuery);
```

Reply:
I like it!